### PR TITLE
Feature/NQ-1342: Simple python register map exporter

### PR DESCRIPTION
**Pull request by Jonah-Foley**
https://nu-quantum.youtrack.cloud/issue/NQ-1342/Simple-register-map-python-exporter

### CODE REVIEW

**Comment by Jonah-Foley** [comment]
Example output for the `simple8` register map:

```python
# Copyright Nu Quantum Ltd
# SPDX-License-Identifier: MIT
#
# Auto-generated by peakrdl-sv

class Simple8RegMap:

    # Register address offsets

    MY_REGFILE_0_CONTROL = 0x0

    MY_REGFILE_0_STATUS = 0x1

    MY_REGFILE_0_CLKDIV = 0x2

    MY_REGFILE_1_CONTROL = 0x3

    MY_REGFILE_1_STATUS = 0x4

    MY_REGFILE_1_CLKDIV = 0x5

    MY_PTIME = 0xC

    MY_STB = 0xD

    MY_WIDE_0 = 0x10
    MY_WIDE_1 = 0x11

    MY_ARRAY_0_0 = 0x20
    MY_ARRAY_0_1 = 0x21

    MY_ARRAY_1_0 = 0x22
    MY_ARRAY_1_1 = 0x23

    MY_ARRAY_2_0 = 0x24
    MY_ARRAY_2_1 = 0x25

    MY_ARRAY_3_0 = 0x26
    MY_ARRAY_3_1 = 0x27

    MY_DATA_0 = 0x30

    MY_DATA_1 = 0x31

    MY_DATA_EXT = 0x40

    MY_WIDE_2_0 = 0x50
    MY_WIDE_2_1 = 0x51

    CONST = 0x60

    MY_PTIME_HIGH = 0xFF8

    MY_PTIME_HIGH1 = 0xFFC
```

File: src/peakrdl_sv/__peakrdl__.py [hunk 1]
```diff
@@ -5,39 +5,38 @@
...
+class VerilogExporter(ExporterSubcommandPlugin):
     short_desc = "A SystemVerilog SystemRDL exporter"
 
     def add_exporter_arguments(self, arg_group: argparse.ArgumentParser) -> None:
         arg_group.add_argument(
             "--reset-polarity",
             default="high",
-            type=validate_reset_polarity,
+            type=str,
+            choices=["high", "low"],
```
Comment (Jonah-Foley) [inline src/peakrdl_sv/__peakrdl__.py:20]: Unrelated to the PR but noticed that this could be simplified, unless the original approach was deliberate.

Comment (ghost) [reply src/peakrdl_sv/__peakrdl__.py:20]: That looks fine so you can remove the `validate*` method.

Comment (Jonah-Foley) [reply src/peakrdl_sv/__peakrdl__.py:20]: Done, that change was also included in this PR.

**Review by ghost — APPROVED** [review]

**Pull request merged**